### PR TITLE
NewItemFactory parameter added

### PR DESCRIPTION
**Pull request by Luk164**
- I added a parameter for NewItemFactory. This will be used instead of the default creator if specified (not null).

### CODE REVIEW

**Comment by Luk164** [comment]
Related to issue #1464

File: .gitignore [hunk 1]
```diff
@@ -243,4 +243,5 @@ ModelManifest.xml
 
 # jekyll
 _site
-Gemfile.lock
\ No newline at end of file
+Gemfile.lock
+*.dt
```
Comment (stsrki) [inline .gitignore:247]: Is this added by mistake?

File: Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs [hunk 2]
```diff
@@ -958,6 +958,11 @@ protected IEnumerable<TItem> DisplayData
         /// </summary>
         [Parameter] public Action<TItem> NewItemDefaultSetter { get; set; }
 
+        /// <summary>
+        /// Function that, if set, is called to create new instance of an item. If left null a default constructor will be used.
+        /// </summary>
+        [Parameter] public Func<TItem> NewItemFactory { get; set; }
```
Comment (stsrki) [inline Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs:964]: Maybe name it `NewItemCreator`? A _factory_ kinda points to a similar design pattern but it doesn't feel like it in this case.

File: .gitignore (hunk 1 above)
Comment (Luk164) [reply .gitignore:247]: Yeah, sorry. That is from a time tracking plugin. Feel free to discard.

File: Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs (hunk 2 above)
Comment (Luk164) [reply Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs:964]: @stsrki Maybe... Factory normally creates more than one type so you may be right. It does not matter to me in the end as long as I can specify it.

Comment (Luk164) [reply Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs:964]: Also I I am not sure if we should pass a parameter with the method. It is possible to pass up to 16 params but that would require a LOT of overloads so a single parameter which holds a class with other parameters would be the preferred approach.

Comment (Luk164) [reply Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs]: Changed to creator in new commit.

File: .gitignore (hunk 1 above)
Comment (Luk164) [reply .gitignore]: Removed in next commit. No idea Where the GemFile.lock came from so I left it there.

**Pull request merged**